integrationtests/tests/fixtures/real.rs

```diff
@@ -10,59 +8,104 @@ use async_trait::async_trait;
 use bitcoin::{secp256k1, Address, Transaction, Txid};
 use bitcoincore_rpc::{Client, RpcApi};
 use cln_rpc::model::requests;
+use cln_rpc::primitives::{Amount as ClnRpcAmount, AmountOrAny};
 use cln_rpc::{ClnRpc, Request, Response};
 use fedimint_bitcoind::DynBitcoindRpc;
 use fedimint_core::encoding::Decodable;
 use fedimint_core::module::registry::ModuleDecoderRegistry;
 use fedimint_core::Amount;
 use fedimint_testing::btc::BitcoinTest;
 use fedimint_wallet_server::common::txoproof::TxOutProof;
-use futures::lock::Mutex;
 use lazy_static::lazy_static;
 use lightning_invoice::Invoice;
+use tokio::sync::Mutex;
 use tokio::time::sleep;
-use tonic_lnd::lnrpc::Invoice as LndInvoice;
-use tonic_lnd::{connect, LndClient};
+use tonic_lnd::lnrpc::{GetInfoRequest, Invoice as LndInvoice, ListChannelsRequest};
+use tonic_lnd::LndClient;
+use tracing::info;
 use url::Url;
 
+use super::GatewayNode;
 use crate::fixtures::LightningTest;
 
 #[derive(Clone)]
 pub struct RealLightningTest {
     rpc_cln: Arc<Mutex<ClnRpc>>,
     rpc_lnd: Arc<Mutex<LndClient>>,
     initial_balance: Amount,
+    // Which lightning node to use as the gateway?
+    gateway_node: GatewayNode,
     pub gateway_node_pub_key: secp256k1::PublicKey,
 }
 
 #[async_trait]
 impl LightningTest for RealLightningTest {
     async fn invoice(&self, amount: Amount, expiry_time: Option<u64>) -> Invoice {
-        let mut lnd_rpc = self.rpc_lnd.lock().await;
-        let tonic_invoice = match expiry_time {
-            Some(expiry) => LndInvoice {
-                value_msat: amount.msats as i64,
-                expiry: expiry as i64,
-                ..Default::default()
-            },
-            None => LndInvoice {
-                value_msat: amount.msats as i64,
-                ..Default::default()
-            },
-        };
-        let invoice_resp = lnd_rpc
-            .lightning()
-            .add_invoice(tonic_invoice)
-            .await
-            .unwrap()
-            .into_inner();
-
-        Invoice::from_str(&invoice_resp.payment_request).unwrap()
+        match self.gateway_node {
+            // If we're using CLN as the gateway, use LND to fetch an invoice
+            GatewayNode::Cln => {
+                info!("fetching invoice from lnd");
+                let mut lnd_rpc = self.rpc_lnd.lock().await;
+                let tonic_invoice = match expiry_time {
+                    Some(expiry) => LndInvoice {
+                        value_msat: amount.msats as i64,
```

Comment:
Strange....lnd allows value_msat to be negative?

Reply:
indeed, that's what they have in spec: https://github.com/lightningnetwork/lnd/blob/master/lnrpc/lightning.proto#L3377

opportunity for LND contrib?